serverless.yaml

```diff
@@ -1,362 +1,417 @@
+  logs:
+    restApi:
+      executionLogging: true
+      fullExecutionData: true
+  tracing:
+    apiGateway: true
+
+functions:
+  fhirServer:
+    timeout: 40
+    description: 'FHIR API Server'
+    events:
+      - http:
+          authorizer:
+            type: COGNITO_USER_POOLS
+            authorizerId:
+              Ref: ApiGatewayAuthorizer
+            scopes: # must have both scopes
+              - 'openid'
+              - 'profile'
+              - 'aws.cognito.signin.user.admin'
+          method: ANY
+          path: '/{proxy+}'
+          private: true
+    handler: src/index.default
+    provisionedConcurrency: 5
+
+  ddbToEs:
+    timeout: 300
+    runtime: nodejs12.x
+    description: 'Write DDB changes from `resource` table to ElasticSearch service'
+    role: DdbToEsLambdaRole
+    handler: ddbToEsLambda/index.handler
+    events:
+      - stream:
+          type: dynamodb
+          arn:
+            'Fn::GetAtt':
+              - ResourceDynamoDBTableV2
+              - StreamArn
+          batchSize: 100
+          maximumRetryAttempts: 3
+          startingPosition: LATEST
+
+resources:
+  - Description: (SO0128) - Solution - Master Template - This template creates all the necessary resources to deploy FHIR Works on AWS; a framework to deploy a FHIR server on AWS.
+  - Parameters:
+      ShouldCreateManagementUser:
+        Type: String
+        Default: true
+        Description: "Set to 'true' to create management IAM user, 'false' if providing user ARN. Default: true"
+        AllowedValues: [ true, false ]
+      UserPassword:
+        Type: String
+        Description: Password for development IAM user. Ignored if ShouldCreateManagementUser is false.
+        NoEcho: true
+      ExistingManagementUserARN:
+        Type: String
+        Default: ""
+        Description: The ARN of the IAM user to use for KMS key management. Ignored if ShouldCreateManagementUser is true.
+      Stage:
+        Type: String
+        Default: ${self:custom.stage}
+        Description: "The deployment stage (e.g. dev, qa, prod). Default: dev"
+      KibanaUserEmail:
+        Type: String
+        AllowedPattern: ^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$
+        Description: E-mail address of the Kibana user. Ignored if Stage is not 'dev'.
+      CognitoOAuthCallbackURL:
+        Type: String
+        Default: ${self:custom.oauthCallback}
+        Description: "Cognito OAuth callback URL used for User Pool. Default:  ${self:custom.oauthCallback}"
+      CognitoOAuthRedirectURL:
+        Type: String
+        Default: ${self:custom.oauthRedirect}
+        Description: "Cognito OAuth redirect URL used for User Pool. Default:  ${self:custom.oauthRedirect}"
+  - Conditions:
+      isDev: !Equals [!Ref Stage, 'dev']
+      isNotDev: !Not [Condition: isDev]
+      CreateManagementUser: !Equals [!Ref ShouldCreateManagementUser, true]
+  - Resources:
+      ManagementUser:
```

Comment:
Just to understand a bit more what is the goal of having this user/why did you add it? is it a best practice?

Reply:
My vote is to remove this and have the customer's be responsible for the management aspect. @nguyen102 @carvantes  any opinion here?

Reply:
Either way the customer needs to provide the template with an IAM User Arn, right? If we can automate that by having this `ManagementUser`, it seems that this would make the installation process simpler for the customer.

Reply:
I think this management user is purely for managing the created AWS resources from this template, not for the deployment. If it was required for deployment it would need to be created prior to the cfn template

Reply:
Looks like this IAM user is used to manage the KMS keys and to allow customers running `sls offline` to [access ES](https://github.com/awslabs/fhir-works-on-aws-deployment/blob/5780f97afe427b26405d82062fbc86b7f61d43a8/cloudformation/elasticsearch.yaml#L149). 

I guess it would be simpler if we ask the user to provide the IAM user they're using to deploy the FHIR server, instead of creating a new IAM user for them.

Reply:
Do not create the `ManagementUser`.

Also, the KMS/ES resource based policies don't really need to mention `devAwsUserAccountArn`/`ManagementUser`. They are sort of redundant. Not sure about exactly why such statement was added in the first place.

If you are deploying/offline-testing FWoA with a role user/role that has and identity police that allows `kms:*`, `es:*` (as suggested on the iam_policy.json file) then removing the above KMS/ES statements has no effect.